### PR TITLE
feat: add function to validate autoshard content topic

### PR DESCRIPTION
**Pull request by adklempner**
## Problem

There is currently no function for validating content topics used for autosharding

## Solution

 

This PR adds a function to sharding utils which validates that a string is a valid content topic in the context of autosharding

## Notes

- Related to #1710

### CODE REVIEW

File: packages/utils/src/common/sharding.spec.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
+import { expect } from "chai";
```
Comment (weboko) [inline packages/utils/src/common/sharding.spec.ts:1]: You can keep this test in `packages/tests` similarly how other utils are done there.
https://github.com/waku-org/js-waku/blob/master/packages/tests/tests/utils.spec.ts

Comment (adklempner) [reply packages/utils/src/common/sharding.spec.ts:1]: @weboko I tired this but now it takes a lot longer to run the unit tests for this function. IMO it shouldn't share a test file that needs docker running in order to not fail.

File: packages/utils/src/common/sharding.ts [hunk 2]
```diff
@@ -18,3 +18,42 @@ export function ensurePubsubTopicIsConfigured(
     );
   }
 }
+
+/**
+ * Given a string, will throw an error if it is not formatted as a valid content topic for autosharding based on https://rfc.vac.dev/spec/51/
+ * @param contentTopic String to validate
+ */
+export function validateContentTopic(contentTopic: string): void {
```
Comment (fryorcraken) [inline packages/utils/src/common/sharding.ts:26]: nitpick: when the method throws an error on failed check, then I believe the `ensure` prefix is more commonly used.

**Review by fryorcraken — APPROVED** [review]

File: packages/utils/src/common/sharding.ts (hunk 2 above)
Comment (adklempner) [reply packages/utils/src/common/sharding.ts]: renamed to `ensureValidContentTopic`

**Pull request merged**